python_modules/dagster/dagster/_core/definitions/declarative_scheduling/scheduling_evaluation_info.py

```diff
@@ -13,10 +13,30 @@
     AssetConditionEvaluationState,
     AssetSubsetWithMetadata,
 )
+from dagster._core.definitions.metadata import MetadataMapping
 from dagster._model import DagsterModel
 from dagster._utils import utc_datetime_from_timestamp
 
 
+class AssetSliceWithMetadata(NamedTuple):
```

Comment:
Ok we can workshop this name later and generalize it but this is a good intermediate step.